### PR TITLE
feat: add backoff delay settings for error code 429

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Description
If error code 420 is encountered in Config and Ping, will retry request using exponential backoff with initial delay of 60s

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by MdUsmanGani — APPROVED** [review]
LGTM

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/ConfigWorker.kt [hunk 1]
```diff
@@ -84,7 +85,8 @@ internal class ConfigWorker(
             // Schedule a ping request to message mixer. Initial delay is 0
             messagePingScheduler.pingMessageMixerService(0)
             Timber.tag(TAG).d("Config Response: %b", response.body()?.data?.enabled)
-        } else return if (response.code() >= HttpURLConnection.HTTP_INTERNAL_ERROR) {
+        } else return if (response.code() == InAppMessagingConstants.RETRY_ERROR_CODE ||
+                response.code() >= HttpURLConnection.HTTP_INTERNAL_ERROR) {
             // Retry with exponential backoff if server has error.
             Result.retry()
```
[the remark this answers is not in the document]
Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/ConfigWorker.kt:91]: updated handling to add random values

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/schedulers/ConfigScheduler.kt [hunk 2]
```diff
@@ -4,41 +4,45 @@ import androidx.work.ExistingWorkPolicy
...
         }
 
         /**
          * This method syncs with config service in the background with all necessary input data
          * bundled into Data and passed into Worker.
          */
-        private fun getConfigWorkRequest(): OneTimeWorkRequest? =
+        private fun getConfigWorkRequest(delay: Long): OneTimeWorkRequest? =
                 // Creating a config work request.
                 OneTimeWorkRequest.Builder(ConfigWorker::class.java)
```
Comment (imaginaris) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/schedulers/ConfigScheduler.kt:44]: what about `.setBackoffCriteria`? Do we want to use 10s (for non 429 errors)  just like on iOS?

Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/schedulers/ConfigScheduler.kt:44]: The default setting is used for other errors, which is exponential backoff and initial delay of 30s.

Comment (imaginaris) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/schedulers/ConfigScheduler.kt:44]: Ok so let's leave it for now.

**Review by imaginaris — APPROVED** [review]

**Pull request merged**